packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1048,6 +1064,204 @@ describe("ModularChangeFamily", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const withBuilds = addMissingBuilds(makeAnonChange(input), getDetachedNode, [
+				u1,
+				a2,
+				c1,
+			]);
+			assert.deepEqual(withBuilds, expected);
+		});
+
+		it("preserves irrelevant builds that are present in the input", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const withBuilds = addMissingBuilds(makeAnonChange(input), getDetachedNode, []);
+			assert.deepEqual(withBuilds, expected);
+		});
+
+		describe("attempts to add relevant builds that are missing from the input", () => {
+			it("adds the missing build if the detached node is available", () => {
+				const input: ModularChangeset = {
+					fieldChanges: new Map([]),
+				};
+
+				const expected: ModularChangeset = {
+					fieldChanges: new Map([]),
+					builds: new Map([
+						[
+							aMajor,
+							new Map([
+								[brand(1), node1Chunk],
+								[brand(2), node2Chunk],
+							]),
+						],
+						[bMajor, new Map([[brand(1), node3Chunk]])],
+					]),
+				};
+
+				const withBuilds = addMissingBuilds(makeAnonChange(input), getDetachedNode, [
+					a1,
+					a2,
+					b1,
+				]);
+				assert.deepEqual(withBuilds, expected);
+			});
+
+			it("throws if the detached node is not available", () => {
+				const input: ModularChangeset = {
+					fieldChanges: new Map([]),
+				};
+
+				assert.throws(() =>
+					addMissingBuilds(makeAnonChange(input), getDetachedNode, [{ minor: 2 }]),
+				);
+			});
+		});
+	});
+
+	describe("filters builds", () => {
+		const aMajor = mintRevisionTag();
+		const bMajor = mintRevisionTag();
+		const a1 = { major: aMajor, minor: 1 };
+		const b1 = { major: bMajor, minor: 1 };
+
+		const node2 = singleJsonCursor(2);
+		const node2Chunk = treeChunkFromCursor(node2);
+		const node3 = singleJsonCursor(3);
+		const node3Chunk = treeChunkFromCursor(node3);
+
+		it("preserves relevant builds that are present in the input", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
```

Comment:
I think we're missing a test for the case where a relevant build is not present in the input (i.e., "tolerates missing relevant builds".).